### DISCUSSION TITLE
[과목 3] 1장 수행 구조 연습 문제

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#28] [연습문제]
### 연습 문제

**문제 1. 백그라운드 프로세스에 대한 설명으로 맞지 않는 것은?**

1) Oracle PMON 프로세스와 SQL Server OPS 쓰레드는 장애가 발생한 시스템을 재기동할 때 인스턴스 복구를 수행한다.
2) Oracle DBWn 프로세스와 SQL Server Lazywriter 쓰레드는 Dirty 버퍼를 데이터 파일에 기록하는 역할을 맡는다.
3) Checkpoint는 이전에 Checkpoint가 일어났던 마지막 시점 이후의 데이터베이스 변경 사항을 데이터 파일에 기록하도록 트리거링한다.
4) Log Writer는 로그 버퍼를 로그 파일에 기록하는 역할을 맡는다.

**문제 2. 보기 중 SQL 특징과 가장 거리가 먼 것은?**

1) 구조적
2) 선언적
3) 집합적
4) 절차적

**문제 3. 오라클이 사용자로부터 전달받은 SQL을 처리하는 과정에 참여하는 서브 엔진과 가장 거리가 먼 것은?**

1) SQL Parser
2) Row-Source Generator
3) Process Monitor
4) Optimizer

**문제 4. SQL Server에서 힌트를 기술하는 3가지 방법으로 틀린 것은?**

1) 테이블 힌트
2) 조건절 힌트
3) 쿼리 힌트
4) 조인 힌트

**문제 5. 아래는 I/O 효율화 튜닝 방안을 설명하고 있다. 잘못된 것은?**

1) 필요한 최소 블록만 읽도록 쿼리를 작성한다.
2) 전략적인 인덱스 구성은 물론 DBMS가 제공하는 다양한 기능을 활용한다.
3) 옵티마이저 행동에 영향을 미치는 가장 중요한 요소는 통계정보이므로 변경이 거의 없는 테이블일지라도 매일 통계정보를 수집해 준다.
4) SQL 조정만으로 좋은 실행계획을 얻기 어렵다면 옵티마이저 힌트를 사용한다.

**Comment by beolggoolosori** [comment]
1. **1** ( SMON(System Monitor) 프로세스와 SQL Server의 Recovery 쓰레드가 인스턴스 복구를 수행)
2. **4**
3. **3** (Process Monitor(PMON)는 백그라운드 프로세스)
4. **2**
5. **3**
